pkg/controllers/clustergroup/controller.go

```diff
@@ -69,6 +69,14 @@ func (h *handler) OnClusterChange(key string, cluster *fleet.Cluster) (*fleet.Cl
 		if sel.Matches(labels.Set(cluster.Labels)) {
 			h.clusterGroups.Enqueue(cg.Namespace, cg.Name)
 		}
+		// if cluster is removed from CG, need to reconcile if ClusterCount doesnt match
+		clusters, err := h.clusterCache.List(cg.Namespace, sel)
+		if err != nil {
+			logrus.Errorf("error fetching clusters in clustergroup %s%s: %v", cg.Namespace, cg.Name, err)
+		}
```

Comment:
@StrongMonkey wonder if we should account for "IsNotFound" here...

Reply:
List is not going to return notFound error I think